File: Java/프로그래머스/1/12903. 가운데 글자 가져오기/가운데 글자 가져오기.java
class Solution {
    public String solution(String s) {
        String answer = "";
        int center = s.length() % 2 == 0 ? (int) (s.length() / 2 - 1) : (int) Math.floor(s.length() / 2);
        if (s.length() % 2 == 0) {
            answer = s.substring(center, center + 2);
        } else {
            answer = s.substring(center, center + 1);
        }
        return answer;
    }
}

File: Kotlin/프로그래머스/1/12903. 가운데 글자 가져오기/가운데 글자 가져오기.kt
class Solution {
    fun solution(s: String): String {
        var answer = ""
        var idx = 0
        if (s.length % 2 == 0) {
            idx = s.length / 2 - 1
            answer = s.substring(idx, idx + 2)
        } else {
            idx = s.length / 2
            answer = s.substring(idx, idx + 1)
        }
        return answer
    }
}